File: Singleton Pattern/Singleton.java


class Sinlgeton {

    static Sinlgeton obj = new Sinlgeton();

    private Sinlgeton() {

        System.out.println("In Private Singleton Cons");
    }

    static Sinlgeton getInstance() {

        return obj;
    }   
}

class Demo {
    public static void main(String[] args) {
        
        Sinlgeton obj1 = Sinlgeton.getInstance();
        Sinlgeton obj2 = Sinlgeton.getInstance();   

        System.out.println(System.identityHashCode(obj1));
        System.out.println(System.identityHashCode(obj2));
    }
}

File: Singleton Pattern/singleton.dart
class Demo {
  static Demo obj = new Demo._private();

  Demo._private() {
    print("In private constructor Demo");
  }

  factory Demo() {
    return obj;
  }
}

void main() {
  Demo obj = Demo.obj;
  print(obj);

  Demo obj2 = Demo.obj;
  // print(obj2.toString());

  
}
